src/Ethereum/ABI/ParamFactory.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "ParamFactory.h"
+#include "HexCoding.h"
+
+using namespace std;
+
+namespace TW::Ethereum::ABI {
+
+std::shared_ptr<ParamBase> ParamFactory::make(const std::string& type) {
+    shared_ptr<ParamBase> param;
+    if (type == "address") {
+        param = make_shared<ParamAddress>();
+    } else if (type == "uint256") {
+        param = make_shared<ParamUInt256>();
+    } else if (type == "bool") {
+        param = make_shared<ParamBool>();
+    } else if (type == "bytes") {
+        param =  make_shared<ParamByteArray>();
+    } else if (type == "string") {
+        param = make_shared<ParamString>();
+    }
+    return param;
```

Comment:
What if type is none of the above?

Reply:
Will only add most common used types first, what else should we add in first batch?

Reply:
most types are handled now